### PR TITLE
Adds prettier and tslint and runs them as a precommit hook

### PR DESCRIPTION
**Pull request by thgreasi**
Borrowed these from the Dashboard.

Change-Type: patch

### CODE REVIEW

**Comment by thgreasi** [comment]
This tries to be pretty close to the [TypeScript_Coding_Guide](https://github.com/resin-io/process/blob/master/process/TypeScript_Coding_Guide.md) and that's why the configurations contain so many options.
I was surprised that the current index.ts used spaces instead of tabs :confused:

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "jsdoc-format": true,
+    "whitespace": [
+      false,
+      "check-type"
+    ],
+    "object-literal-sort-keys": false,
+    "only-arrow-functions": [false],
+    "quotemark": [true, "single", "avoid-escape"],
+    "max-classes-per-file": [false],
+    "max-line-length": [180],
```
Comment (CameronDiver) [inline tslint.json]: This should be 120 AFAIK. My understanding is soft limit of 80 chars, hard limit of 120.

Comment (thgreasi) [reply tslint.json]: :+1: changing that... Surprisingly prettier uses 120!

**Comment by thgreasi** [comment]
No versionbot here :(

**Comment by hedss** [comment]
@thgreasi Wish, and it shall appear! Now VB covered. :)

**Comment by thgreasi** [comment]
Woohoo!

**Pull request merged**